src/SizeAndPositionManager.ts

```diff
@@ -51,18 +54,67 @@ export default class SizeAndPositionManager {
       this.estimatedItemSize = estimatedItemSize;
     }
 
-    if (itemSizeGetter != null) {
-      this.itemSizeGetter = itemSizeGetter;
+    if (itemSize != null) {
+      this.itemSize = itemSize;
+    }
+
+    this.checkForItemSizeOverflow();
+
+    if (this.justInTime() && this.totalSize != null) {
+      this.totalSize = undefined;
+    } else {
+      this.computeTotalSizeAndPositionData();
+    }
+  }
+
+  checkForItemSizeOverflow() {
```

Comment:
Yeah, didn't feel completely right yet, but felt weird duplicating, if it's outside the class it will need all of the properties to be passed. Not sure what is best practice with this kind of error handling.

Are you thinking of something like?

* `checkForMismatchItemSizeAndItemCount`